cpp/src/io/avro/reader_impl.cu

```diff
@@ -235,28 +245,29 @@ rmm::device_buffer reader::impl::decompress_data(const rmm::device_buffer& comp_
   return decomp_block_data;
 }
 
-void reader::impl::decode_data(const rmm::device_buffer& block_data,
-                               const std::vector<std::pair<uint32_t, uint32_t>>& dict,
-                               device_span<string_index_pair> global_dictionary,
-                               size_t num_rows,
-                               std::vector<std::pair<int, std::string>> selection,
-                               std::vector<column_buffer>& out_buffers,
-                               rmm::cuda_stream_view stream)
+void decode_data(metadata* meta,
+                 const rmm::device_buffer& block_data,
+                 const std::vector<std::pair<uint32_t, uint32_t>>& dict,
+                 device_span<string_index_pair> global_dictionary,
+                 size_t num_rows,
+                 std::vector<std::pair<int, std::string>> selection,
+                 std::vector<column_buffer>& out_buffers,
+                 rmm::cuda_stream_view stream)
```

Comment:
```suggestion
void decode_data(metadata const* meta,
                 rmm::device_buffer const& block_data,
                 std::vector<std::pair<uint32_t, uint32_t>> const& dict,
                 device_span<string_index_pair> global_dictionary,
                 size_t num_rows,
                 std::vector<std::pair<int, std::string>> const& selection,
                 std::vector<column_buffer>& out_buffers,
                 rmm::cuda_stream_view stream)
```

Reply:
In addition, please avoid having any output parameter. If possible then use this:
```suggestion
std::vector<column_buffer> decode_data(metadata const* meta,
                 rmm::device_buffer const& block_data,
                 std::vector<std::pair<uint32_t, uint32_t>> const& dict,
                 device_span<string_index_pair> global_dictionary,
                 size_t num_rows,
                 std::vector<std::pair<int, std::string>> const& selection,
                 rmm::cuda_stream_view stream)
```

Reply:
unfortunately metadata and datasource cannot be const.